From 1328f32b3edc2ad9764495afc0429952af2334d0 Mon Sep 17 00:00:00 2001
From: Biwen Li <biwen.li@nxp.com>
Date: Thu, 27 Jun 2019 14:24:29 +0800
Subject: [PATCH 550/741] arm64: dts: ls1028a: Add ftm_alarm0 DT node

commit e5130a2bec505cb0fe29817b9986d393709e09aa from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

The patch adds ftm_alarm0 DT node for LS1028ARDB board
FlexTimer1 module is used to wakeup the system

Signed-off-by: Biwen Li <biwen.li@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
index a09911e88e8a..d29281de6752 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
@@ -17,6 +17,10 @@
 	#address-cells = <2>;
 	#size-cells = <2>;
 
+	aliases {
+		rtc1 = &ftm_alarm0;
+	};
+
 	cpus {
 		#address-cells = <1>;
 		#size-cells = <0>;
@@ -746,6 +750,19 @@
 				reg = <0x000100 0 0 0 0>;
 			};
 		};
+
+                rcpm: rcpm@1e34040 {
+                        compatible = "fsl,ls1028a-rcpm", "fsl,qoriq-rcpm-2.1+";
+                        reg = <0x0 0x1e34040 0x0 0x1c>;
+                        #fsl,rcpm-wakeup-cells = <7>;
+                };
+ 
+                ftm_alarm0: timer@2800000 {
+                        compatible = "fsl,ls1028a-ftm-alarm";
+                        reg = <0x0 0x2800000 0x0 0x10000>;
+                        fsl,rcpm-wakeup = <&rcpm 0x0 0x0 0x0 0x0 0x4000 0x0 0x0>;
+                        interrupts = <GIC_SPI 44 IRQ_TYPE_LEVEL_HIGH>;
+                };
 	};
 
         display0: malidp@f080000 {
-- 
2.17.1

